src/main/java/dude/component/IStorage.java

```diff
@@ -0,0 +1,22 @@
+package dude.component;
+
+public interface IStorage {
```

Comment:
The naming of this interface is rather confusing, as it is not immediately clear what the "I" in IStorage stands for. Alternatives could be `StorageInterface`, or simply just `Storage`.